staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -436,6 +436,59 @@ func TestReflectorListAndWatchInitConnBackoff(t *testing.T) {
 	}
 }
 
+type fakeBackoff struct {
+	clock clock.Clock
+	calls int
+}
+
+func (f *fakeBackoff) Backoff() clock.Timer {
+	f.calls++
+	return f.clock.NewTimer(time.Duration(0))
+}
+
+func TestBackoffOnTooManyRequests(t *testing.T) {
```

Comment:
does this exercise both code paths introduced above?

Reply:
Yes - the case 0 exercises the path of error from the watch requests, case 1 (in the switch in watch func) exercises the "watch event of type error".